Marlin/src/feature/leds/pca9632.cpp

```diff
@@ -89,21 +92,25 @@ static void PCA9632_WriteRegister(const byte addr, const byte regadd, const byte
   Wire.endTransmission();
 }
 
-static void PCA9632_WriteAllRegisters(const byte addr, const byte regadd, const byte vr, const byte vg, const byte vb) {
+static void PCA9632_WriteAllRegisters(const byte addr, const byte regadd, const byte vr, const byte vg, const byte vb
+        #if ENABLED(PCA9632_RGBW)
+          , const byte vw
+        #endif
+                                                                                                                    ) {
   #if DISABLED(PCA9632_NO_AUTO_INC)
     uint8_t data[4], len = 4;
     data[0] = PCA9632_AUTO_IND | regadd;
     data[1 + (PCA9632_RED >> 1)] = vr;
     data[1 + (PCA9632_GRN >> 1)] = vg;
     data[1 + (PCA9632_BLU >> 1)] = vb;
   #else
-    uint8_t data[6], len = 6;
-    data[0] = regadd + (PCA9632_RED >> 1);
-    data[1] = vr;
-    data[2] = regadd + (PCA9632_GRN >> 1);
-    data[3] = vg;
-    data[4] = regadd + (PCA9632_BLU >> 1);
-    data[5] = vb;
+    PCA9632_WriteRegister(addr, regadd + (PCA9632_RED >> 1), vr);
```

Comment:
why did you change to `PCA9632_WriteRegister` instead of the `data[0] = regadd + (PCA9632_RED >> 1);`, as it was before?

Reply:
For some reason this didn't work. It weren't work even with original code but as with default defines in this code this part is never used (as only first 3 channels are used and in this case part with auto increment is used). Even when I try use only RGB lights but I redefine some color to 4th channel, it stops working. With calling PCA9632_WriteRegister for each channel it works. Not sure why... According to some I2C examples between each register should be some verification byte or something so may this cause this issue...